Shared/Coordinator/MainCoordinator.swift

```diff
@@ -31,19 +31,31 @@ class MainCoordinator: Coordinator<Void> {
         UserNotificationManager.shared.delegate = self
         LaunchManager.shared.delegate = self
 
-        self.runLaunchFlow()
+        Task {
+            await self.runLaunchFlow()
+        }
     }
 
-    private func runLaunchFlow() {
-        LaunchManager.shared.launchApp(with: self.launchOptions)
-        self.router.setRootModule(self.splashVC, animated: false)
+    private func runLaunchFlow() async {
+        await self.router.setRootModule(self.splashVC, animated: false)
+
+        let launchStatus = await LaunchManager.shared.launchApp(with: self.launchOptions)
```

Comment:
As mentioned above, this launch function is async so we don't need to use a delegate call to know when it's done. Instead we can simply "await".